### PR TITLE
Fixed: Show more monitors in CNS on very large displays #6503

### LINKED ISSUE #6503: Show more monitors in CNS on very large displays
**Describe the bug**
When CNS is opened in large 4K displays, 75% of the screen space is wasted. 

**To Reproduce**

1. Visit https://care.ohc.network/facility/0c95c7f0-e1d2-4aff-83fa-933cef60d3a8/cns on a 4K display

**Expected behavior**
On large displays, render more vitals monitors into view per page.

**Screenshots**

image

### PR DESCRIPTION
**Pull request by i0am0arunava**
## Proposed Changes

- Fixes #6503

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [X] Ensure that UI text is kept in I18n files.
- [X] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [X] Request for Peer Reviews
- [ ] Completion of QA

## Summary by CodeRabbit

- **New Features**
	- Enhanced responsiveness of the Central Nursing Station component by dynamically adjusting the number of items displayed per page based on window size.
- **Bug Fixes**
	- Improved pagination functionality to better accommodate various screen widths.
- **Style**
	- Updated grid layout class for improved adaptability to different display sizes.

### CODE REVIEW

File: src/components/Facility/CentralNursingStation.tsx [hunk 1]
```diff
@@ -41,11 +40,19 @@ interface Props {
 }
 
 export default function CentralNursingStation({ facilityId }: Props) {
+  let PER_PAGE_LIMIT;
```
Comment (shivankacker) [inline src/components/Facility/CentralNursingStation.tsx]: Can we use a state or a ref instead of let? Seems a little unsafe.

**Review by shivankacker — CHANGES_REQUESTED** [review]

File: src/components/Facility/CentralNursingStation.tsx [hunk 2]
```diff
@@ -43,9 +43,20 @@ interface Props {
 export default function CentralNursingStation({ facilityId }: Props) {
   const { t } = useTranslation();
   const [isFullscreen, setFullscreen] = useFullscreen();
+  const [PER_PAGE_LIMIT, setPerPageLimit] = useState(6);
```
Comment (i0am0arunava) [inline src/components/Facility/CentralNursingStation.tsx:46]: @shivankacker ,is this okay now?

**Review by i0am0arunava — COMMENTED** [review]
@shivankacker Used state instead, is it okay

**Comment by i0am0arunava** [comment]
@rithviknishad ,is it okay now
Screenshot 2024-11-06 195148

**Comment by rithviknishad** [comment]
Closing as duplicate of #9013 and #9028; avoid opening new PR's. resolve conflicts on same PRs. Hard to track requested changes.

File: src/components/Facility/CentralNursingStation.tsx (hunk 2 above)
Comment (rithviknishad) [reply src/components/Facility/CentralNursingStation.tsx:46]: like already said in previous PR's review, use `useBreakpoints` instead. please do not skip reviews by opening fresh PRs

https://github.com/ohcnetwork/care_fe/pull/9028#discussion_r1831129265

Comment (rithviknishad) [reply src/components/Facility/CentralNursingStation.tsx:46]: cc: @bodhish

**Pull request closed without merging**